moveit_core/robot_model/include/moveit/robot_model/joint_model.h

```diff
@@ -350,18 +350,12 @@ class JointModel
 
   /** \brief Get the factor that should be applied to the value returned by distance() when that value is used in
    * compound distances */
-  double getDistanceFactor() const
-  {
-    return distance_factor_;
-  }
-
+  double getDistanceFactor() const = 0;
```

Comment:
why did you make this a pure virtual function?

Reply:
Because I have defined the function in the joint_model.cpp so I did not want to redefine it in the header file

Reply:
```suggestion
  double getDistanceFactor();
```